realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/ProcessCommitTests.java

```diff
@@ -16,159 +16,202 @@
+    @RunTestWithRemoteService(SimpleCommitRemoteService.class)
+    @RunTestInLooperThread
+    public void expectSimpleCommit() {
+        remoteService.createHandler(Looper.myLooper());
+
+        final SyncUser user = UserFactory.getInstance().createDefaultUser(Constants.AUTH_URL);
+        String realmUrl = Constants.SYNC_SERVER_URL;
+        final SyncConfiguration syncConfig = configFactory.createSyncConfigurationBuilder(user, realmUrl).build();
+        final Realm realm = Realm.getInstance(syncConfig);
+        final RealmResults<ProcessInfo> all = realm.where(ProcessInfo.class).findAll();
+        looperThread.keepStrongReference(all);
+        all.addChangeListener(new RealmChangeListener<RealmResults<ProcessInfo>>() {
             @Override
-            public void run() {
-                try {
-                    Looper.prepare();
-                    Context targetContext = InstrumentationRegistry.getTargetContext();
-
-                    SyncUser user = UserFactory.createDefaultUser(Constants.AUTH_URL);
-                    String realmUrl = Constants.SYNC_SERVER_URL;
-                    final SyncConfiguration syncConfig = new SyncConfiguration.Builder(user, realmUrl)
-                            .name(SendOneCommit.class.getSimpleName())
-                            .errorHandler(new SyncSession.ErrorHandler() {
-                                @Override
-                                public void onError(SyncSession session, ObjectServerError error) {
-                                    fail("Sync failure: " + error);
-                                }
-                            })
-                            .build();
-                    Realm.deleteRealm(syncConfig);//TODO do this in Rule as async tests
-                    final Realm realm = Realm.getInstance(syncConfig);
-                    Intent intent = new Intent(targetContext, SendOneCommit.class);
-                    targetContext.startService(intent);
-                    final RealmResults<ProcessInfo> all = realm.where(ProcessInfo.class).findAll();
-                    all.addChangeListener(new RealmChangeListener<RealmResults<ProcessInfo>>() {
-                        @Override
-                        public void onChange(RealmResults<ProcessInfo> element) {
-                            assertEquals(1, all.size());
-                            assertEquals("Background_Process1", all.get(0).getName());
-                            testFinished.countDown();
-                        }
-                    });
-
-                    Looper.loop();
-
-                } catch (Throwable e) {
-                    exception[0] = e;
-                    testFinished.countDown();
-                }
+            public void onChange(RealmResults<ProcessInfo> element) {
+                assertEquals(1, all.size());
+                assertEquals("Background_Process1", all.get(0).getName());
+                realm.close();
```

Comment:
Can these clean up methods somehow be moved to the Rule? These tests are so complex that anything we can do to reduce it would be nice.